### PR TITLE
Compiler perf improvements

### PR DESCRIPTION
**Pull request by mmoskal**
WIP

@riknoll any clue if caching type on Node is a good idea? `(node as any).typeOverride = r`

### CODE REVIEW

**Comment by riknoll** [comment]
It's fine if we don't reuse the TS program (which the compiler does not right now). Might cause issues if we ever decide to do that though, because it tries to keep source files (with the AST) around.

File: cli/cli.ts [hunk 1]
```diff
@@ -3751,6 +3751,9 @@ function buildCoreAsync(buildOpts: BuildCoreOptions): Promise<pxtc.CompileResult
                 }
             }
 
+            if (pxt.appTarget.compile.switches.time)
+                console.log(compileResult.times)
```
Comment (pelikhan) [inline cli/cli.ts]: you can use "pxt.debug"

Comment (mmoskal) [reply cli/cli.ts]: pxt.debug get lost in all other noise, I guess pxt.log would be good, but need to stringify first

Comment (pelikhan) [reply cli/cli.ts]: Not worth it.

**Comment by mmoskal** [comment]
should be good to merge @pelikhan

**Comment by pelikhan** [comment]
@riknoll ?

**Comment by riknoll** [comment]
👍

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**